src/utils/headers.js

```diff
@@ -3,45 +3,127 @@ const fs = require('fs')
+   *
+   * /a/b/c -> ['a', 'b', 'c']
+   */
+  const ruleParts = rule.split('/').filter(Boolean)
+  const pathParts = path.split('/').filter(Boolean)
+  /**
+   * Fast path for /*.
+   */
+  const matchAll = ruleParts.length === 1 && ruleParts[0] === '*'
+  if (matchAll && path) return true
+  /**
+   * If either set of path parts is empty (indicating root dir /), they must
+   * both be empty (/ and /), otherwise there is no match.
+   */
+  const noRuleParts = ruleParts.length === 0
+  const noPathParts = pathParts.length === 0
+  if (noRuleParts || noPathParts) return noRuleParts && noPathParts
+  /**
+   * Otherwise, iterate over ruleParts and pathParts.
+   */
+  for (let index = 0; index < ruleParts.length; index++) {
+    /**
+     * Select the corresponding pathPart for this rulePart.
+     */
+    const rulePart = ruleParts[index]
+    const pathPart = pathParts[index]
+    /**
+     * Whether or not all ruleParts and pathParts have been iterated over.
+     */
+    const noMoreRuleParts = index >= ruleParts.length
+    const noMorePathParts = index >= pathParts.length
+    /**
+     * What kind of placeholder, if any, the current rulePart is.
+     */
+    const ruleIsAsterisk = rulePart === '*'
+    const ruleIsPlaceholder = rulePart.startsWith(':')
+    const ruleIsWildcard = ruleIsAsterisk || ruleIsPlaceholder
+    /**
+     * If all pathParts and ruleParts have been exhausted, the pattern cannot
+     * match the path unless the next rulePart is a trailing wildcard token (*).
+     *
+     * This will catch the case where there are more ruleParts than pathParts as
+     * well, i.e. /static/* and /static.
+     */
+    if (noMorePathParts && noMoreRuleParts) {
+      /**
+       * If the rulePart is the * wildcard, this is a guaranteed match;
+       * otherwise, it's a guaranteed mismatch.
+       */
+      return ruleIsAsterisk
+    }
+    /**
+     * A match-all wildcard (*) will indicate a match for the entire pattern if
+     * it matches final pathPart (meaning all pathParts now match) OR if the
+     * wildcard is the final rulePart, i.e. `/path/to/*` matched against
+     * `/path/to/file/etc`.
+     */
+    if (ruleIsAsterisk && (noMorePathParts || noMoreRuleParts)) return true
+    /**
+     * For either either wildcard (* or :placeholder):
+     */
+    if (ruleIsWildcard) {
+      if (noMorePathParts) {
+        /**
+         * The entire path is a definite match if the wildcard matches the final
+         * pathPart, i.e. `/path/to/:placeholder` and `/path/to/*` matched
+         * against `/path/to/something`.
+         */
+        return true
       }
-    } else {
-      return rulePart === target
+    } else if (rulePart !== pathPart) {
+      /**
+       * Finally, if the parts do not match, the rule does not match.
+       */
+      return false
     }
   }
-
-  return false
+  /**
+   * If no mismatch was found, the rule matches the target path.
+   */
+  return true
 }
```

Comment:
Rewrote `matchPaths` logic to accept `*` wildcard at places other than the final `rulePart`.